File: src/SeoToolkit.Umbraco.Redirects.Core/Models/Business/Redirect.cs
using System;
using System.Text.RegularExpressions;
using Umbraco.Cms.Core.Models;
using Umbraco.Cms.Core.Models.PublishedContent;
using Umbraco.Cms.Core.Routing;
using Umbraco.Extensions;
using SeoToolkit.Umbraco.Redirects.Core.Extensions;
using Umbraco.Cms.Core.Models.Membership;

namespace SeoToolkit.Umbraco.Redirects.Core.Models.Business
{
    public class Redirect
    {
        public int Id { get; set; }
        public bool IsEnabled { get; set; }
        public bool IsRegex { get; set; }
        public Domain Domain { get; set; }
        public string CustomDomain { get; set; }
        public string OldUrl { get; set; }
        public string NewUrl { get; set; }
        public IPublishedContent NewNode { get; set; }
        public ILanguage NewNodeCulture { get; set; }
        public DateTime LastUpdated { get; set; }
        public int CreatedBy { get; set; }
        public int RedirectCode { get; set; }

        public string GetNewUrl()
        {
            return NewUrl.IfNullOrWhiteSpace(NewNode?.Url(NewNodeCulture?.IsoCode?.ToLowerInvariant()));
        }
    }
}

File: src/SeoToolkit.Umbraco.Redirects/assets/src/models/Redirect.ts
export interface Redirect { 
    id: number;
    unique: string;
    entityType: string;
    isEnabled: boolean;
    isRegex: boolean;
    domain?: number | null;
    customDomain?: string | null;
    oldUrl?: string | null;
    newUrl?: string | null;
    newNodeId?: string | null;
    newCultureIso?: string | null;
    redirectCode: number;
    lastUpdated?: (string) | null;
}